### PR TITLE
feat/default-lookup-services-consumerGroups

### PR DESCRIPTION
**Pull request by AntoineJac**
### Summary

feat/default-lookup-services-consumerGroups
Similar changes to: https://github.com/Kong/go-database-reconciler/pull/44

PR: https://konghq.atlassian.net/browse/FTI-6181

Related to https://github.com/Kong/deck/pull/1367

+ fix for consumers groups:
https://konghq.atlassian.net/browse/FTI-6185

### Full changelog

* [Implement ...]
* [Fix ...]

### Issues resolved

Fix #XXX

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/Kong/docs.konghq.com/pull/XXX)

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [ ] Manual testing on Kubernetes

### CODE REVIEW

**Review by Prashansa-K — DISMISSED** [review]

**Review by mheap — APPROVED** [review]

File: pkg/dump/dump.go [hunk 1]
```diff
@@ -251,7 +291,30 @@ func getProxyConfiguration(ctx context.Context, group *errgroup.Group,
 		if err != nil {
 			return fmt.Errorf("plugins: %w", err)
 		}
+
+		if config.LookUpSelectorTagsConsumerGroups != nil {
+			globalConsumerGroupsPlugins, err := GetAllPlugins(ctx, client, config.LookUpSelectorTagsConsumerGroups)
+			if err != nil {
+				return fmt.Errorf("error retrieving global plugins: %w", err)
+			}
+			// if globalServices are not present, add them.
```
Comment (GGabriele) [inline pkg/dump/dump.go:300]: bad copy & paste

File: pkg/file/builder.go [hunk 2]
```diff
@@ -886,7 +902,23 @@ func (b *stateBuilder) ingestService(s *FService) error {
...
+	sort.Strings(stringTags)
+	sort.Strings(b.lookupTagsServices)
+	// if the service tags and the lookup tags are the same, it means
+	// that the service is a global service retrieved from upstream,
+	// therefore we don't want to merge its tags with the selected tags.
+	if !reflect.DeepEqual(stringTags, b.lookupTagsServices) {
+		utils.MustMergeTags(&s.Service, b.selectTags)
+	}
+
+	// utils.MustMergeTags(&s, b.selectTags)
```
Comment (GGabriele) [inline pkg/file/builder.go:921]: Should remove this?

File: pkg/dump/dump.go (hunk 1 above)
Comment (AntoineJac) [reply pkg/dump/dump.go]: It is fixed thanks

File: pkg/dump/dump.go [hunk 3]
```diff
@@ -251,7 +291,30 @@ func getProxyConfiguration(ctx context.Context, group *errgroup.Group,
 		if err != nil {
 			return fmt.Errorf("plugins: %w", err)
 		}
+
+		if config.LookUpSelectorTagsConsumerGroups != nil {
```
[the remark this answers is not in the document]
Comment (AntoineJac) [reply pkg/dump/dump.go]: Correct I removed this, not sure why I added it... 
I have tested and confirm behaviour is ok

File: pkg/file/builder.go [hunk 4]
```diff
@@ -177,7 +179,21 @@ func (b *stateBuilder) consumerGroups() {
 				cg.ID = kong.String(*current.ID)
 			}
 		}
-		utils.MustMergeTags(&cg.ConsumerGroup, b.selectTags)
+
```
[the remark this answers is not in the document]
Comment (AntoineJac) [reply pkg/file/builder.go:182]: Yes correct, see above

**Review by GGabriele — APPROVED** [review]

**Pull request merged**